### PR TITLE
Edited process for creating output directories to avoid race condition

### PR DESCRIPTION
**Pull request by dominicanstey**
Changed the process for making output directories to a 'try/except', to avoid the race condition created by the 'if not' process

### CODE REVIEW

**Comment by williamjameshandley** [comment]
Hi @dominicanstey 

Unfortunately the tests are showing that whilst this fixes the race condition for the MPI case, it breaks for the non-mpi case. 

Do you need the conditional statement now? Couldn't you just run os.makedirs on all cores regardless of rank in this setup?

You should also specify that you're catching the OSError exception, rather than a generic  one.

**Comment by williamjameshandley** [comment]
@dominicanstey any update on the above?

File: pypolychord/__init__.py [hunk 1]
```diff
@@ -176,11 +176,17 @@ def run_polychord(loglikelihood, nDims, nDerived, settings,
...
 
-    if not os.path.exists(settings.base_dir) and rank == 0:
-        os.makedirs(settings.base_dir)
-
-    if not os.path.exists(settings.cluster_dir) and rank == 0:
-        os.makedirs(settings.cluster_dir)
+    try:
+        if rank == 0:
+            os.makedirs(settings.base_dir)
+    except:
```
Comment (williamjameshandley) [inline pypolychord/__init__.py]: Could you change this to a specific exception, e.g. OSError

File: pypolychord/__init__.py [hunk 2]
```diff
@@ -176,11 +176,17 @@ def run_polychord(loglikelihood, nDims, nDerived, settings,
...
+        pass
+        
+    try:
+        if rank == 0:
+            os.makedirs(settings.cluster_dir)
+    except:
```
Comment (williamjameshandley) [inline pypolychord/__init__.py]: Same here

Comment (dominicanstey) [reply pypolychord/__init__.py]: Done. Sorry, I forgot to change that the first time.

**Pull request merged**